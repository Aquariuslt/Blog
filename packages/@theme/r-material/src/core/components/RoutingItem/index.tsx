import * as React from 'react';
import { Link as RouterLink } from 'react-router-dom';
import { makeStyles, StyleRules, Theme, useTheme } from '@material-ui/core/styles';

import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import ListItem from '@material-ui/core/ListItem';

import { NavMenuItem } from '../../stores/navigation.store';
import BundledIcon from '../../components/BundledIcon';

type RoutingItemProps = NavMenuItem;

const useStyles = makeStyles(
  (theme: Theme): StyleRules => ({
    root: {
      display: 'flex',
      flexDirection: 'column',
      width: 360
    },
    lists: {
      backgroundColor: theme.palette.background.paper
    },
    nested: {
      paddingLeft: theme.spacing(4)
    }
  })
);

const RoutingItem: React.ComponentType<RoutingItemProps> = (props: RoutingItemProps): JSX.Element => {
  const theme: Theme = useTheme();
  const classes = useStyles(theme);

  const isExtLink = props.ext;

  const listItemInnerRenderProps = {
    component: RouterLink,
    to: props.link
  };

  const listItemExtRenderProps = {
    component: 'a',
    href: props.link,
    target: '_blank',
    rel: 'noopener'
  };

  const actualRenderProps = isExtLink ? listItemExtRenderProps : listItemInnerRenderProps;

  return (
    <ListItem button className={classes.nested} {...actualRenderProps}>
      <ListItemIcon>
        <BundledIcon type="starBorder" />
      </ListItemIcon>
      <ListItemText primary={props.name} />
    </ListItem>
  );
};

export default RoutingItem;
