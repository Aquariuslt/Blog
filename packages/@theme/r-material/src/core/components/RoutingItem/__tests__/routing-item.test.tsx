import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

import RoutingItem from '../';

Enzyme.configure({ adapter: new Adapter() });

describe('@theme/r-material: core/components: RoutingItem', (): void => {
  it('# should mount routing item component correctly', (): void => {
    const mockName = 'routing menu';
    const mockLink = '/to/somewhere';

    const wrapper = Enzyme.shallow(<RoutingItem name={mockName} link={mockLink} />);

    expect(wrapper).toMatchSnapshot();
  });
});
