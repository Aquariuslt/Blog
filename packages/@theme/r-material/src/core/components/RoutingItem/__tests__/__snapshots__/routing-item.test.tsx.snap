// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@theme/r-material: core/components: RoutingItem # should mount routing item component correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  className="makeStyles-nested-3"
  component={[Function]}
  to="/to/somewhere"
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <BundledIcon
      type="starBorder"
    />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    primary="routing menu"
  />
</WithStyles(ForwardRef(ListItem))>
`;
